### DISCUSSION TITLE
Disabling parser rules

### DISCUSSION THREAD

**Discussion by silverpill** [kivikakk/comrak#436] [Feature requests]
Is it possible to skip parsing of certain elements, such as headings or lists?
In my application I need to support only a subset of GFM. The workaround I'm currently using is re-rendering of certain AST nodes back to markdown and replacing them with text nodes, but perhaps there's a better way?

**Comment by kivikakk** [comment]
There is no such feature at present, though I imagine it might not be too hard to add checks at key places in the parser where certain elements are recognised. Pull requests welcome!

**Comment by silverpill** [comment]
>converted this issue into discussion

I'm still interested in this.

I imagine that instead of re-rendering back to markdown (lossy), one can use sourcepos to get the original text, but when I tried to do that I found that some sourcepos have unexpected values. Don't remember what exactly was wrong, maybe positions were set to zeros

**Comment by davemackintosh** [comment]
I would love to see this ability, for my software I want to be able to turn off the indentation to code rule so only fenced code is rendered as code as I use this as part of the render pipeline in [weaving](https://github.com/davemackintosh/weaving) where partials will obviously contain indentation for the elements but are rendered as code currently. I will take a look to see if I can figure out a way to do this.

**Reply by kivikakk** [reply]
Yepyep, understood. Feel free to drop any ideas you come up with here to check if they sound reasonable, or just open a PR with what works for you.

It may also be worth considering if you can achieve what you want with a custom formatter HTML formatter, or even simple text manipulation on the input before feeding it in — stripping all leading whitespace in the input, with a simple state machine to avoid doing so in ```` ```fenced ```` code blocks. If you can keep it simple and reliable, it can sometimes be easier than changing the parser itself.

> Closest I found was Hugo and then I got to thinking- I haven't seen that film with Hugo Weaving in it where he wears a dress for ages.

Is that The Adventures of Priscilla, Queen of the Desert?
